### ISSUE TITLE
Samba smb:// mounts don't mount again once unmounted

### ISSUE THREAD

**Issue by qgates** [linuxmint/nemo#2049]
```
 * Nemo 4.0.6
 * Both windowed and non-windowed
 * Distribution - Linux Mint 19.1
 * Intel HD 5500 (Thinkpad T450)
 * 64 bit
 ```

**Issue**

Mounting a samba share in Nemo everything works as expected. Unmounting the share and remounting it again results in the error "Unknown file type" displayed in a modal when trying to show the contents of the share. In this case the shares are password protected (ie. not anonymous) but the option 'remember password until logout' is selected upon the first successful connect.

This issue dates back to nemo 2.x, remains unresolved, and was originally reported on launchpad (see other info below).

**Steps to reproduce**

1. Type smb:// share into nemo location, ie. smb://servername/share
2. Enter credentials and choose 'remember until logout'
3. Everything works as expected, ie. share location is displayed and we can work with files/folders
4. Unmount the share using nemo sidebar eject button or other method
5. Repeat step 1
6. Dialog with error "Unknown file type" appears and share contents are not displayed

**Expected behaviour**

Nemo should cleanly disconnect from and reconnect to smb:// locations without issue.

**Workarounds**

1. Kill nemo once the shares are unmounted:
- $ nemo -q (fixes issues accessing share via sidebar)
- $ nemo-desktop -q (fixes issue accessing share via auto appearing desktop shortcut, if enabled)
2. Logout and login again
3. Once the error has occurred I can successfully mount the share with its IP, ie. smb://192.168.0.xx/share. But if I unmount and attempt to reconnect via IP, I see the same problem

**Other information**

1. Everything tested above is on the same LAN and subnet
2. The problem can also be caused by using an external tool to open a smb:// intent. For example, transmission-remote-gui to open a torrent's location, if the location's share has not been previously mounted in nemo

This longstanding bug dates back to at least 2015 and was originally reported on Launchpad:
https://bugs.launchpad.net/linuxmint/+bug/1401754

**Comment by momo42** [comment]
I'm also affected on two different PCs, also Mint 19.1. Interestingly I noticed this very annoying bug just a few weeks ago. I thought I ruined my Nemo while upgrading from Mint 19 to 19.1 but today I saw it on a freshly installed Mint 19.1 machine. So I decided to visit the issue tracker.

Thank you so much for your kill workaround! So I don't need to logout login every time. :-)

I also found this old 2015 bug but I guess these bugs are not the same. Didn't see this behavior before 2019 and I'm using smb shares all the time.

**Comment by jbalazsi** [comment]
I can reproduce this issue on LinuxMint 19.1, 19.2 and 19.3 as well. This is the reason I am using the Mate edition of LinuxMint.

**Comment by rlazoryshchak** [comment]
I've faced with this issue too

**Comment by DevTheNew** [comment]
The program [Gigolo](https://packages.ubuntu.com/bionic/gigolo) offers a workaround. If I unmount a network drive with it, then I can mount it again in nemo without any problems.

**Comment by icarter09** [comment]
Nemo 4.8.2
Mint 20 64bit

Verified this is still an issue.

**Comment by spsf64** [comment]
I just noticed this bug still present in nemo 4.8.4 running Manjaro Cinnamon edition.
Any workaround?
TIA

**Comment by lefti696** [comment]
Same here on Debian testing and Nemo 4.8.4

**Comment by leigh123linux** [comment]
Try navigating the open window to network:/// before unmounting and see if it reproduces.

Maybe something like this would fix the issue.

https://github.com/GNOME/nautilus/commit/0852847e2613dbcebb4ed7f58e5b2aee3aa86a90
https://github.com/GNOME/nautilus/commit/a23ed4ad0173ca7a51e26282b85d06ce4b25912a

**Comment by icarter09** [comment]
Verified that navigating the open window to network:/// before unmounting does not reproduce the issue.

**Comment by JpegXguy** [comment]
I think either this issue or #2411 should be merged/one of them declared duplicate. It's useful to have the discussion in one place. Nice find, @leigh123linux . It works for me!

**Comment by icarter09** [comment]
As long as you select any other window location other than the Samba location you plan to unmount, it should be able to remount successfully. Can someone on this ticket verify if Nautilus has this same issue? Thanks.

**Comment by JpegXguy** [comment]
You are correct. It seems browsing to any location NOT under the smb directory tree  will work. Nautilus doesn't have this problem. In fact it will be able to mount these network shares even after Nemo has declared that the location is not a folder

**Comment by icarter09** [comment]
@JpegXguy thanks for update about Nautilus.

**Comment by jmarewski** [comment]
I can confirm this is still an issue in Nemo 5.4.3

**Comment by voxspox** [comment]
Still an issue in Nemo 5.6.4

As mentioned above selecting any other location before unmounting helps.

**Comment by Fantu** [comment]
Today after saw [this report](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1059418) I did a search and arrived here.
From some fast tests I was unable to reproduce this issue with nemo 5.8.5 on debian unstable.
I don't understand if is solved in nemo or is it related to other component and solved in newer version of it in debian unstable

**Comment by claudiux** [comment]
@Fantu That's right! This bug seems to have been resolved with nemo 5.8.5.

**Comment by Fantu** [comment]
Can you please tell me which commit has solved it? In a related issue, a commit was mentioned as a possible fix, but it was only applied for 6.0.0
